component/site/libraries/dbmodel.php

```diff
@@ -2580,7 +2580,20 @@ function listIcalEventsByRange($startdate, $enddate, $limitstart, $limit, $showr
 		{
 			$query .="\n GROUP BY rpt.rp_id";
 		}
-
+/*
+		//Dirty hack to include locations categories grouping
+		if(strpos($query, "loccatmap"))
+		{
+			if(strpos($query, 'GROUP BY'))
+			{
+				$query .= ", loc.loc_id";
+			}
+			else
+			{
+				$query .= "GROUP BY loc.loc_id";
+			}
+		}
+*/
```

Comment:
Carlos - I don't think this would work where there the events are already grouped by repeat or event since the group by location id is redundant?